include/secp256k1_silentpayments.h

```diff
@@ -165,6 +168,256 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_recipien
+
+/** Create Silent Payment shared secret.
+ *
+ *  Given the public input data (secp256k1_silentpayments_recipient_public_data),
+ *  and the recipient's 32 byte scan key, calculate the shared secret.
+ *
+ *  The resulting shared secret is needed as input for creating silent payments
+ *  outputs belonging to the same recipient scan public key. This function is
+ *  intended for light clients, i.e., scenarios where the caller does not have
+ *  access to the full transaction. If the caller does have access to the full
+ *  transaction, `secp256k1_silentpayments_recipient_scan_outputs` should be
+ *  used instead.
+ *
+ *  Returns: 1 if shared secret creation was successful. 0 if an error occurred.
+ *  Args:                  ctx: pointer to a context object
+ *  Out:       shared_secret33: pointer to the resulting 33-byte shared secret
+ *  In:   recipient_scan_key32: pointer to the recipient's 32 byte scan key
+ *                 public_data: pointer to the public_data object, loaded using
+ *                              `_recipient_public_data_parse`
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_recipient_create_shared_secret(
+    const secp256k1_context *ctx,
+    unsigned char *shared_secret33,
+    const unsigned char *recipient_scan_key32,
+    const secp256k1_silentpayments_recipient_public_data *public_data
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
+
+/** Create Silent Payment output public key.
+ *
+ *  Given a shared_secret, a public key B_spend, and an output counter k,
+ *  create an output public key.
+ *
+ *  This function is used by the recipient when scanning for outputs without
+ *  access to the transaction outputs (e.g., using BIP158 block filters). When
+ *  scanning with this function, it is the scanners responsibility to determine
+ *  if the generated output exists in a block before proceeding to the next
+ *  value of `k`.
+ *
+ *  Returns: 1 if output creation was successful. 0 if an error occurred.
+ *  Args:                   ctx: pointer to a context object
+ *  Out:         P_output_xonly: pointer to the resulting output x-only pubkey
+ *  In:         shared_secret33: shared secret, derived from either sender's
+ *                               or recipient's perspective with routines from
+ *                               above
+ *       recipient_spend_pubkey: pointer to the recipient's spend pubkey
+ *                               (labeled or unlabeled)
+ *                            k: output counter (initially set to 0, must be
+ *                               incremented for each additional output created
+ *                               or after each output found when scanning)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_recipient_create_output_pubkey(
+    const secp256k1_context *ctx,
+    secp256k1_xonly_pubkey *P_output_xonly,
```

Comment:
nit: could maybe consider removing the `P_` prefix as we don't use this naming scheme anywhere else in this or other modules (I think I introduced that in the first take of the BIP352 module for some reason)